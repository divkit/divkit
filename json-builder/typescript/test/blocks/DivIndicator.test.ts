import {
    divCard,
    DivIndicator,
    DivRoundedRectangleShape,
    fixed,
    IDivData,
    reference,
    rewriteRefs,
    templateHelper,
} from '../../src';

describe('DivIndicator tests', (): void => {
    it('should create simple block', (): void => {
        const block = new DivIndicator({
            pager_id: 'pager1',
            active_item_color: '#fff',
            inactive_item_color: '#000',
            space_between_centers: fixed(1),
            shape: new DivRoundedRectangleShape({
                item_height: fixed(20, 'sp'),
                item_width: fixed(100),
                corner_radius: fixed(10, 'dp'),
            }),
            animation: 'worm',
        });

        expect(block).toMatchSnapshot();
    });

    it('should create card with templates', (): void => {
        const templates = {
            div: new DivIndicator({
                pager_id: reference('pager_id'),
                shape: new DivRoundedRectangleShape({
                    item_height: reference('shape_height'),
                }),
            }),
        };

        const divData: IDivData = {
            log_id: 'id',
            states: [
                {
                    state_id: 1,
                    div: templateHelper(templates).div({
                        pager_id: 'pager1',
                        shape_height: fixed(20),
                        animation: 'slider',
                    }),
                },
            ],
        };

        expect(divCard(rewriteRefs(templates), divData)).toMatchSnapshot();
    });
});
