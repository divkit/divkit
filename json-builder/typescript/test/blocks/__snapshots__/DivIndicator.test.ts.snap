// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DivIndicator tests should create card with templates 1`] = `
{
  "templates": {
    "div": {
      "type": "indicator",
      "shape": {
        "type": "rounded_rectangle",
        "$item_height": "shape_height"
      }
    }
  },
  "card": {
    "log_id": "id",
    "states": [
      {
        "state_id": 1,
        "div": {
          "type": "div",
          "pager_id": "pager1",
          "shape_height": {
            "type": "fixed",
            "value": 20
          },
          "animation": "slider"
        }
      }
    ]
  }
}
`;

exports[`DivIndicator tests should create simple block 1`] = `
{
  "type": "indicator",
  "active_item_color": "#fff",
  "animation": "worm",
  "inactive_item_color": "#000",
  "pager_id": "pager1",
  "shape": {
    "type": "rounded_rectangle",
    "corner_radius": {
      "type": "fixed",
      "unit": "dp",
      "value": 10
    },
    "item_height": {
      "type": "fixed",
      "unit": "sp",
      "value": 20
    },
    "item_width": {
      "type": "fixed",
      "value": 100
    }
  },
  "space_between_centers": {
    "type": "fixed",
    "value": 1
  }
}
`;
